stacker/blueprints/base.py

```diff
@@ -402,6 +406,44 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def parse_user_data(self, raw_user_data):
+        """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+         Args:
+            raw_userdata (str of userdata):
+                str of the given userdata to be parsed
+
+        """
+        pattern = re.compile(r'{{([::|\w]+)}}')
+        user_data = []
+        start_index = 0
+        variables = self.get_variables()
+        parameters = self.get_cfn_parameters()
+
+        for match in pattern.finditer(raw_user_data):
```

Comment:
1. I decided to stick to just using a regex because the standard delimiter in the Template strings is $ which is very commonly used in UserData for other purposes. We can subclass the Template class and define our own custom rules for the delimiters, but I would argue that what we have right now is cleaner.

2. Parameters are in get_variables, however, they are stored as CFNParameter objects. In my new commit I addressed this is a cleaner way.